services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/attribute/EnumAttributeConverter.java

```diff
@@ -41,23 +48,30 @@
     private final Class<T> enumClass;
     private final Map<String, T> enumValueMap;
 
-    private EnumAttributeConverter(Class<T> enumClass) {
+    private final Function<T, String> keyExtractor;
+
+    private EnumAttributeConverter(Class<T> enumClass, Function<T, String> keyExtractor) {
         this.enumClass = enumClass;
+        this.keyExtractor = keyExtractor;
 
         Map<String, T> mutableEnumValueMap = new LinkedHashMap<>();
         Arrays.stream(enumClass.getEnumConstants())
-              .forEach(enumConstant -> mutableEnumValueMap.put(enumConstant.toString(), enumConstant));
+              .forEach(enumConstant -> mutableEnumValueMap.put(keyExtractor.apply(enumConstant), enumConstant));
 
         this.enumValueMap = Collections.unmodifiableMap(mutableEnumValueMap);
     }
 
     public static <T extends Enum<T>> EnumAttributeConverter<T> create(Class<T> enumClass) {
-        return new EnumAttributeConverter<>(enumClass);
+        return new EnumAttributeConverter<>(enumClass, Enum::toString);
+    }
+
+    public static <T extends Enum<T>> EnumAttributeConverter<T> createWithNameAsKeys(Class<T> enumClass) {
+        return new EnumAttributeConverter<>(enumClass, Enum::name);
```

Comment:
I think this means that the default converter that virtually everyone will use (by virtue of using DefaultAttributeConverterProvider) will continue using the unsafe toString() version. That's fine if that's the way you all want to go, but I figured I'd point it out.

Reply:
Exactly, this is for backward compatibility.